src/pelagic_base.F90

```diff
@@ -215,6 +251,55 @@ function get_sinking_rate(self,_ARGUMENTS_LOCAL_) result(rm)
 
       rm = self%rm
    end function
+   
+   subroutine DO (self, _ARGUMENTS_DO_)
+      class (type_ersem_pelagic_base), intent(in) :: self
+      
+      real (rk) :: k1_rate,state, et, ETW
+      
+      
+      _DECLARE_ARGUMENTS_DO_
+      
+         _LOOP_BEGIN_
+         ! first order kinetics
```

Comment:
Some more detail on this could be added (2-3 sentences, confirming that the same decay rate is applied to all constituents, and example of potential use).

Reply:
comments have been added to explain the new lines